clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy

```diff
@@ -132,22 +150,33 @@ class ServerGroupParameters {
    * @return
    */
   static List<String> unescapePythonUnicodeJsonList(String string) {
-    string?.split(",")?.collect { s ->
+    List result = string?.split(",")?.collect { s ->
       s.replace("u'", "").replace("'", "").replace("[", "").replace("]", "").replaceAll("([ ][ ]*)", "")
     } ?: []
+    return result
   }
 
   /**
-   * Stack parameters of type 'comma_delimited_list' come back as a unicode json string. We need to split that up.
+   * Some stack parameters of type 'json' come back as a unicode json string. We need to split that up.
    *
    * TODO See https://bugs.launchpad.net/heat/+bug/1613415
    *
    * @param string
    * @return
    */
   static Map<String, String> unescapePythonUnicodeJsonMap(String string) {
-    String parsed = string?.replace("u'", "")?.replace("'", "")?.replace("{", "")?.replace("}", "")?.replace("\"", "")?.replaceAll("([ ][ ]*)", "")
-    parsed ? Splitter.on(",").withKeyValueSeparator(":").split(parsed) : [:]
+    String parsed = string
+      ?.replaceAll(':\\p{javaWhitespace}*None\\p{javaWhitespace}*([,}])', ': null$1') // first replace python None with json null
```

Comment:
Python will export unicode json with `u` prefixes? Weird

Reply:
It’s a big in certain versions of openstack where they are effectively calling a “toString()” on a thing that doesn’t have it implemented. So instead of writing out valid json, it writes out this Unicode json junk.

Reply:
Thanks, gents!